src/components/page/ProfilePage.tsx

```diff
@@ -1,10 +1,34 @@
 import { MobileLayout } from '@/components/layouts/MobileLayout';
 import { HeaderMenuCircleButton } from '@/components/parts/Header';
+import { UserProfile } from '@/features/profile/components';
+import { getCurrentUserId } from '@/lib/api/auth/server';
+import { getUserById } from '@/lib/api/user/server';
+
+export const ProfilePage = async () => {
+  /**
+   * Get the user ID of the logged-in user based on the auth token.
+   * Using the user ID, fetch the user's information from the server.
+   * @returns  The user's name on success, undefined on error.
```

Comment:
It does not return `undefined` but throws an Error on failure. Because `unwrap` is only valid when `result.ok` is truthy.

Reply:
Thank you for pointing out! I refined the TSDoc, and also updated the function itself to throw the error declaratively, instead of letting `unwrap()` cause an error.